src/main/java/org/ballerinalang/command/util/OSUtils.java

```diff
@@ -207,14 +207,23 @@ private static boolean isSolaris() {
         return OS.contains("sunos");
     }
 
+    private static boolean isCentOS() {
+        return System.getProperty("user.name").contains("root");
+    }
+
     /**
      * Provide user home directory based on command.
      * @return user home directory
      */
     private static String getUserHome() {
         String userHome = System.getenv("HOME");
         if (userHome == null) {
-            userHome = System.getProperty("user.home");
+            if (isCentOS()) {
```

Comment:
So the logic here should be if user mismatch we should try to derive the user home